### PR TITLE
Feat(LSP): Goto Definition.

### PR DESCRIPTION
**Pull request by He1pa**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y #297  

#### 2. What is the scope of this PR (e.g. component or file name):
LSP

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):
impl goto definition for lsp, support:
1. goto definition of type or value(id) in assign stmt
2. goto definition of schema in unification and assign stmt
3. goto definition of import stmt

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: kclvm/ast/src/ast.rs [hunk 1]
```diff
@@ -213,6 +215,17 @@ impl Program {
             None => vec![],
         }
     }
+
+    pub fn pos_to_stmt(&self, pos: &Position) -> Option<Node<Stmt>> {
```
Comment (Peefy) [inline kclvm/ast/src/ast.rs:219]: Add more documents to explain why it is necessary to obtain AST from pos in reverse and its functions.

File: kclvm/ast/src/ast.rs [hunk 2]
```diff
@@ -237,6 +250,15 @@ impl Module {
         }
         return stmts;
     }
+
+    pub fn pos_to_stmt(&self, pos: &Position) -> Option<Node<Stmt>> {
```
Comment (Peefy) [inline kclvm/ast/src/ast.rs:254]: Add more documents to explain why it is necessary to obtain AST from pos in reverse and its functions.

File: kclvm/tools/src/LSP/src/goto_def.rs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+use kclvm_ast::pos::ContainsPos;
+use kclvm_config::modfile::KCL_FILE_EXTENSION;
+use kclvm_error::Position as KCLPos;
+use kclvm_sema::resolver::scope::ProgramScope;
+use lsp_types::{GotoDefinitionResponse, Url};
+use lsp_types::{Location, Range};
+
+use crate::to_lsp::lsp_pos;
+
+pub(crate) fn goto_definition(
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/goto_def.rs:14]: Public APIs with more documents and tests.

File: kclvm/tools/src/LSP/src/goto_def.rs [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    program: Program,
+    kcl_pos: KCLPos,
+    prog_scope: ProgramScope,
+) -> Option<lsp_types::GotoDefinitionResponse> {
+    match program.pos_to_stmt(&kcl_pos) {
+        Some(node) => match node.node {
+            Stmt::Unification(stmt) => goto_def_for_unification(stmt, kcl_pos, prog_scope),
+            Stmt::Assign(stmt) => goto_def_for_assign(stmt, kcl_pos, prog_scope),
+            Stmt::Import(stmt) => goto_def_for_import(stmt, kcl_pos, prog_scope, program),
+            _ => None,
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/goto_def.rs:24]: Leave TODO?

File: kclvm/tools/src/LSP/src/goto_def.rs [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        },
+        None => None,
+    }
+}
+
+pub(crate) fn find_name_in_program_scope(
+    name: &String,
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/goto_def.rs:31]: s/&String/&str

Use `cargo clippy` to fix more such issues.

File: kclvm/tools/src/LSP/src/goto_def.rs [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+                    scopes.push(c.borrow().clone());
+                }
+            }
+        }
+    }
+    positions
+}
+
+pub(crate) fn positions_to_goto_def_resp(
+    positions: &Vec<(KCLPos, KCLPos)>,
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/goto_def.rs:58]: s/&Vec<(KCLPos, KCLPos)>/&[(KCLPos, KCLPos)]

Use `cargo clippy` to fix more such issues.

Comment (Peefy) [inline kclvm/tools/src/LSP/src/goto_def.rs:58]: What does the KCLPos tuple and Vec mean?

File: kclvm/tools/src/LSP/src/request.rs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    _snapshot: LanguageServerSnapshot,
+    params: lsp_types::GotoDefinitionParams,
+    sender: Sender<Task>,
+) -> anyhow::Result<Option<lsp_types::GotoDefinitionResponse>> {
+    let file = params
+        .text_document_position_params
+        .text_document
+        .uri
+        .path();
+    let pos = params.text_document_position_params.position;
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/request.rs:62]: Use small functions fot line 62-69

File: kclvm/tools/src/LSP/src/goto_def.rs (hunk 6 above)
Comment (He1pa) [reply kclvm/tools/src/LSP/src/goto_def.rs]: tuple means (start, end) and vec means there may be multiple jump targets, such as `import pkg`, you can choose to jump to which file under pkg

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**